### PR TITLE
Replace one more occurence of Tuple used as payload in SPC with KeyValuePair

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Comment by stephentoub** [comment]
The concrete benefit of such a change is if either:
a) No other untrimmed code paths are using `ValueTuple<T1, T2>`, in which case we could trim `ValueTuple<T1, T2>` away in favor keeping `KeyValuePair<T1, T2>`, and that's a better trade because KVP has less surface area
b) No other untrimmed code paths are using `ValueTuple<reference type, reference type>`, in which case an AOT build wouldn't need to stamp the code for that generic specialization
Yes?  Is that going to be the case?

File: src/libraries/System.Private.CoreLib/src/System/Threading/SynchronizationContext.cs [hunk 1]
```diff
@@ -19,7 +21,7 @@ public SynchronizationContext()
 
         public virtual void Send(SendOrPostCallback d, object? state) => d(state);
 
-        public virtual void Post(SendOrPostCallback d, object? state) => ThreadPool.QueueUserWorkItem(static s => s.d(s.state), (d, state), preferLocal: false);
+        public virtual void Post(SendOrPostCallback d, object? state) => ThreadPool.QueueUserWorkItem(static s => s.Key(s.Value), new KeyValuePair<SendOrPostCallback, object?>(d, state), preferLocal: false);
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Threading/SynchronizationContext.cs:24]: This is odd use of KeyValuePair. There is no key-value relationship between the two items.

Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/System/Threading/SynchronizationContext.cs:24]: I agree but `s.d(s.state)` is not descriptive either.

**Comment by marek-safar** [comment]
> Yes? Is that going to be the case?

case a) 

It should be the case today but I agree that probably not in the future. Is it not worth it?

**Pull request closed without merging**